test-case-reporting/test/test_result_record.test.ts

```diff
@@ -0,0 +1,148 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {AssertionError} from 'assert';
+import {DB, KarmaReporter, Travis} from 'test-case-reporting';
+import {Database, dbConnect} from '../src/db';
+import {TestResultRecord} from '../src/test_result_record';
+import {getFixture} from './fixtures';
+import {setupDb} from '../src/setup_db';
+import Knex from 'knex';
+import md5 from 'md5';
+
+jest.mock('../src/db', () => ({
+  dbConnect: () =>
+    Knex({
+      client: 'sqlite3',
+      connection: ':memory:',
+      useNullAsDefault: true,
+    }),
+}));
+
+describe('TestResultRecord', () => {
+  let db: Database;
+  let testResultRecord: TestResultRecord;
+  let sampleBuild: Travis.Build;
+  let sampleJob: Travis.Job;
+  let sampleKarmaReport: KarmaReporter.TestResultReport;
+
+  beforeAll(async () => {
+    db = dbConnect();
+    await setupDb(db);
+
+    testResultRecord = new TestResultRecord(db);
+
+    sampleBuild = (getFixture(
+      'sample-travis-build'
+    ) as unknown) as Travis.Build;
+
+    sampleJob = (getFixture('sample-travis-job') as unknown) as Travis.Job;
+
+    sampleKarmaReport = (getFixture(
+      'sample-karma-report'
+    ) as unknown) as KarmaReporter.TestResultReport;
+  });
+
+  afterAll(() => {
+    db.destroy();
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+    db('test_runs').truncate();
+    db('test_cases').truncate();
+    db('jobs').truncate();
+    db('builds').truncate();
+  });
+
+  describe('insertTravisBuild', () => {
+    it('adds the build to the database', async () => {
+      await testResultRecord.insertTravisBuild(sampleBuild);
+
+      const build = await db<DB.Build>('builds').select().first();
+      expect(build.commit_sha).toEqual(sampleBuild.commitSha);
+      expect(build.build_number).toEqual(sampleBuild.buildNumber);
+    });
+  });
+
+  describe('insertTravisJob', () => {
+    let buildId: number;
+
+    beforeEach(async () => {
+      buildId = await testResultRecord.insertTravisBuild(sampleBuild);
+    });
+
+    it('adds the job to the database if build id is good', () => {
+      testResultRecord.insertTravisJob(
+        {
+          jobNumber: '413413.612',
+          testSuiteType: 'unit',
+        },
+        buildId
+      );
+    });
+
+    it('fails to add the job to the database if build id is bad');
+  });
+
+  describe('testStatus', () => {
+    it('returns SKIP when the test was skipped', () => {
+      expect(testResultRecord.testStatus(true, false)).toEqual('SKIP');
```

Comment:
Nit: when making calls like this where the positional arguments are non-obvious, sometimes we include the names in the comments ie. 
`testResultRecord.testStatus(/* skipped */ true, /* success */ false)`

Reply:
How would you feel about taking in an object that looks like {skipped: true, success: false}, so that the names have to be included (and so we can simply pass the test status)?

Reply:
SGTM, can do the same with makeTestName

Reply:
Nit: Want to have `makeTestName` and `makeTestStatus`, or `testName` and `testStatus`?

Reply:
The latter seems fine

Reply:
Done.